shotover-proxy/tests/codec/util/packet_capture.rs

```diff
@@ -135,41 +127,23 @@ impl PacketCapture {
         file_name: &Path,
         filter: Option<String>,
     ) -> Vec<Result<ParsedPacket, String>> {
-        // TODO: Fix flakiness from out-of-order futures.
-        // let pool = ThreadPool::new(num_cpus::get() * 2);
-        let pool = ThreadPool::new(1);
```

Comment:
Is there anything more to the simplification than getting to use `par_iter()` instead of creating the threadpool and using `pool.execute`?

Reply:
we also ditch the Arc<Mutex<T>>